src/lavinmq/exchange/exchange.cr

```diff
@@ -316,7 +316,7 @@ module LavinMQ
         destination:      @destination.name,
         destination_type: @destination.is_a?(Queue) ? "queue" : "exchange",
         routing_key:      routing_key,
-        arguments:        arguments,
+        arguments:        arguments.nil? ? Hash(String, AMQP::Field).new : arguments,
```

Comment:
Hashes are allocated on the heap, `NamedTuple`'s on the stack. Both becomes `"{}"` when `.to_json`. 

```suggestion
        arguments:        arguments || NamedTuple.new,
```

```
icr:11> Benchmark.memory { 1000.times { Hash(String, String).new } }
 => 61440
icr:12> Benchmark.memory { 1000.times { NamedTuple.new } }
 => 0
icr:13> Hash(String, String).new.to_json
 => "{}"
icr:14> NamedTuple.new.to_json
 => "{}"
```